packages/loopring_v2/contracts/helper/RingHelper.sol

```diff
@@ -171,11 +192,22 @@ library RingHelper {
         internal
         pure
     {
-        ring.valid = ring.valid && (ring.size > 1 && ring.size <= 8); // invalid ring size
+        // NOTICE: deprecated logic, rings must be of size 2 now
+        // ring.valid = ring.valid && (ring.size > 1 && ring.size <= 8); // invalid ring size
+        
+        ring.valid = ring.valid && ring.size == 2;
+
+        // Ring must consist of a buy and a sell
+        ring.valid = ring.valid && (
+            (ring.participations[0].order.isBuy() && !ring.participations[1].order.isBuy()) ||
+            (ring.participations[1].order.isBuy() && !ring.participations[0].order.isBuy())
+        );
```

Comment:
Is this needed? If the order is used as the maker it doesn't really matter if it's a buy or a sell order, so matching buy/buy and sell/sell should be fine. Unless you have other reasons to do this.

Reply:
Its more so an easy way to check that the orders have the same primary token denoted in the `trancheB`. I'm not completely sure that orders need to have the same primary token set, however. @coreycaplan3 any thoughts?

Reply:
Yeah because we need to ensure that both orders have the same primary-ticker (base symbol). That would change how we match things on our end if the primary is not the same